### DISCUSSION TITLE
What's the best way to re-use AsyncImages in LazyColumns?

### DISCUSSION THREAD

**Discussion by vitorpamplona** [coil-kt/coil#2219] [Q&A]
I have a Twitter-like LazyColumn feed where the profile picture of each post can repeat in separate items of the lazy column. On heavy images, it's visible that the AsyncImage re-loads for each for the repeated instances. 

Do you know if having two AsyncImages in separate items creates two identical textures in the GPU for rendering? 

Is there an efficient way to re-use AsyncImage in multiple items? Maybe caching only one painter for each image url in the viewModel? Or is that already handled by AsyncImage directly?

**Comment by anonInode** [comment]
AsyncImages could use the same GPU texture(i.e. AHardwareBuffer, dma-buf) if they retrieve the same Bitmap from coil's memoryCache.
But sadly coil doesn't handle this, see https://github.com/coil-kt/coil/issues/1461
It means you need to add a `Interceptor` to ensure pipeline synchronization for the same image key.
@vitorpamplona
